### ISSUE TITLE
Version 0.99.6 training process is normal, 0.100.3 training process error

### ISSUE THREAD

**Issue by lindadamama** [dotnet/TorchSharp#1037]
<img width="1255" alt="err" src="https://github.com/dotnet/TorchSharp/assets/48863486/d11a4cf5-eabb-4e7c-ba72-957abc4143c7">
[W C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\third_party\nvfuser\csrc\graph_fuser.cpp:108] Warning: operator () profile_node %987 : int[] = prim::profile_ivalue(%dims.39)
 does not have profile information (function operator ())
  

[W C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\third_party\nvfuser\csrc\manager.cpp:340] Warning: FALLBACK path has been taken inside: torch::jit::fuser::cuda::runCudaFusionGroup. This is an indication that codegen Failed for some reason.
To debug try disable codegen fallback path via setting the env variable `export PYTORCH_NVFUSER_DISABLE=fallback`
 (function runCudaFusionGroup)  

[W C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\third_party\nvfuser\csrc\manager.cpp:340] Warning: FALLBACK path has been taken inside: torch::jit::fuser::cuda::runCudaFusionGroup. This is an indication that codegen Failed for some reason.  
  

To debug try disable codegen fallback path via setting the env variable `export PYTORCH_NVFUSER_DISABLE=fallback`
 (function runCudaFusionGroup)
  

2023/7/3 0:23:21
ehope: 0/80000 :
loss:0.048035335
acc:0.9519647
model directory：
【D:\Inno.LabelAssistantDatas\电路板\电路板\AIModel\model.pt】

  

D:\C#\Inno.ImgProcess.UI\Inno.AIFrameWork.TrainTest\bin\Debug\net6.0-windows\Inno.AIFrameWork.TrainTest.exe (process 22168) exited with code -1073741819.
To automatically close the console when debugging stops, enable Tools->Options->Debugging->Automatically close the console when debugging stops.
Press any key to close this window . . .

**Comment by GeorgeS2019** [comment]
@lindadamama 
Because this has Chinese characters, it is important you share in TEXT not in image <=====  

@ChengYen-Tang  Please advice this person so this will not lead to abuse of our kindness

**Comment by GeorgeS2019** [comment]
@lindadamama 
Please take some time to format 
```
[W C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\third_party\nvfuser\csrc\graph_fuser.cpp:108] Warning: operator () profile_node %987 : int[] = prim::profile_ivalue(%dims.39) does not have profile information (function operator ()) [W C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\third_party\nvfuser\csrc\manager.cpp:340] Warning: FALLBACK path has been taken inside: torch::jit::fuser::cuda::runCudaFusionGroup. This is an indication that codegen Failed for some reason. To debug try disable codegen fallback path via setting the env variable `export PYTORCH_NVFUSER_DISABLE=fallback` (function runCudaFusionGroup) [W C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\third_party\nvfuser\csrc\manager.cpp:340] Warning: FALLBACK path has been taken inside: torch::jit::fuser::cuda::runCudaFusionGroup. This is an indication that codegen Failed for some reason. To debug try disable codegen fallback path via setting the env variable `export PYTORCH_NVFUSER_DISABLE=fallback` (function runCudaFusionGroup) 2023/7/3 0:23:21 0/80000次训练: loss(训练集错误率):0.048035335 acc(训练集准确率):0.9519647 模型的保存路径为： 【D:\Inno.LabelAssistantDatas\电路板\电路板\AIModel\model.pt】
D:\C#\Inno.ImgProcess.UI\Inno.AIFrameWork.TrainTest\bin\Debug\net6.0-windows\Inno.AIFrameWork.TrainTest.exe (process 22168) exited with code -1073741819.
To automatically close the console when debugging stops, enable Tools->Options->Debugging->Automatically close the console when debugging stops.
Press any key to close this window . . .
```

**Comment by GeorgeS2019** [comment]
@lindadamama 

Your history
- https://github.com/dotnet/TorchSharp/issues/1017

**Comment by GeorgeS2019** [comment]
To all users here who are sharing in Chinese characters <== Please read this <====

the people who normally reply DO NOT UNDERSTAND Chinese

Please be KIND <=====

**Comment by GeorgeS2019** [comment]
@Oceania2018

I am including here as @lindadamama is a regular user of https://github.com/SciSharp

Please advice @lindadamama  if the people here, often from US,, who would reply to requests involving Chinese Characters, this is really OUT of Kindness

**Comment by lindadamama** [comment]
I'm sorry,I have modified it

**Comment by GeorgeS2019** [comment]
@lindadamama 

Let us all work together with Kindness to keep this community keep on active

**Comment by lindadamama** [comment]
<img width="1267" alt="err3" src="https://github.com/dotnet/TorchSharp/assets/48863486/2ca5bea0-0e2d-4d03-8f7d-9466d6b0753d">
There have been new errors again

**Comment by lindadamama** [comment]
Error message exited with code -1073741819. It should be that Torch2.0.1 occupies more memory than Torch1.13, causing a memory overflow,
Another error is that the Tensor was released due to unknown reasons during the training process of the training set
@GeorgeS2019

**Comment by GeorgeS2019** [comment]
@NiklasGustafsson 

Is the pilar of this project.

I hope when he has time soon to get back to you.

I am independent but someone who believe strongly we need this alternative to pytorch in DeepAI for .Net

**Comment by ChengYen-Tang** [comment]
@lindadamama ,
Can you prepare a simple example and dataset so that everyone can reproduce the problem. You currently provide too little information, it is difficult to analyze the problem.

**Comment by lindadamama** [comment]
I have uploaded the code and there is no problem with version 0.99.6. If 0.100.x doesn't work or maybe your testing is fine,I plan to keep using version 0.99.6
@NiklasGustafsson
@ChengYen-Tang 
@GeorgeS2019 
[ConsoleApp2.zip](https://github.com/dotnet/TorchSharp/files/11968200/ConsoleApp2.zip)

**Comment by lindadamama** [comment]
Sorry, there was an issue with the code just uploaded. It should look like this

````csharp

void checkDiretory(string path) 
{
    if (Directory.Exists(path))
    {
        Directory.Delete(path, true);
    }
    Directory.CreateDirectory(path);
}

````
<img width="313" alt="2023-07-06 211339" src="https://github.com/dotnet/TorchSharp/assets/48863486/79999668-8170-442c-8414-3020c0acbdea">

**Comment by ChengYen-Tang** [comment]
An interesting question!! Without using breakpoints, it crashes in 3 ephos, but with a large number of breakpoints, it can run at least 15 ephos.

I suspect it has something to do with GC.

**Comment by NiklasGustafsson** [comment]
> Please advice @lindadamama if the people here, often from US,, who would reply to requests involving Chinese Characters, this is really OUT of Kindness

I don't mind it if posts include pieces that I cannot understand, but anytime there is important information in Chinese (or any other number of languages that I don't know), I will be either unable to assist, or delayed in doing so. I'm not sure where @GeorgeS2019's concerns about kindness come from, nothing in this post seems unkind to me, but the language issue is definitely a matter of efficiency -- I can only realistically help if the relevant information is in English. Same thing goes for comments in code -- if the comments are not in English, and you upload a repro, it may slow down the process of getting help.

I'm sure there are other community members who can pitch in and help with other languages, but I cannot.

Regarding kindness, and again, _I don't see anything unkind in this post_ -- everyone should recognize that it is in everyone's interest to keep the dialogue professional and to the point. It is very understandable to become frustrated with something that isn't working, or not working as expected, but getting things fixed or answered is always quicker when not venting the frustration publicly.

Thanks,

Niklas

**Comment by lindadamama** [comment]
I will upload a new code that can reproduce the problem, and after you download it, I will delete all the codes
@NiklasGustafsson

**Comment by lindadamama** [comment]
[ConsoleApp2.zip](https://github.com/dotnet/TorchSharp/files/12004142/ConsoleApp2.zip)

**Comment by NiklasGustafsson** [comment]
Okay, I have downloaded it. I will take a look later today.

**Comment by lindadamama** [comment]
It's already 0:20 am on my end and I'm ready to go to bed. If there are any questions, I will reply promptly tomorrow

**Comment by Oceania2018** [comment]
> @Oceania2018
> 
> I am including here as @lindadamama is a regular user of https://github.com/SciSharp
> 
> Please advice @lindadamama if the people here, often from US,, who would reply to requests involving Chinese Characters, this is really OUT of Kindness

Sorry for nothing I can do for this issue due to little knowledge about TorchSharp.

**Comment by NiklasGustafsson** [comment]
@lindadamama -- so, I've tried it now, and I see the following:

1. Everything works fine with 0.99.6

2. When running CUDA, there's some kind of exception deep in the TorchScript runtime, which TorchSharp has no control over. That makes me wonder if this runs perfectly on Python, loading the same model -- do you know? Setting the environment variable gets rid of the warning messages, but not the runtime error.

3. When I disable CUDA, I don't get an exception, but the program just terminates after a few epochs, sometimes during the first. I don't see it using a lot of memory, so it just dies.

The 2.0.1 runtime is different in significant way from 1.13, as I understand it, but

**Comment by lindadamama** [comment]
If the TorchScript runtime is different, I will continue to use the torchsharp0.99.6 version. It is not easy for me to configure a pytorch runtime environment, because
1. The installation package of pytorch is huge. Since the official server of pytorch is not in our country, the installation process is often interrupted. I found a lot of mirror sites to solve the installation problem of pytorch
2. In my python project, in addition to using pytorch, there are some other python library files. These python library files also depend on pytorch. When configuring the environment before, there were many compatibility problems.
3. If the pytorch version is changed, there is a high probability that other libraries will need to be reconfigured, and the code may not run smoothly. I don’t want to do it anymore

**Comment by lindadamama** [comment]
I re-tested with 0.101.5, and the program can run normally
